File: commands/digitalungdom/idea.js
/* global db include guild */

const getUserByDiscordId = include( 'models/get' ).getUserByDiscordId;
const createNotificationEmbed = include( 'utils/embeds/createNotificationEmbed' );

module.exports = {
  name: 'idea',
  description: 'Skicka in ett förslag till Digital Ungdom.',
  aliases: [ 'suggestion', 'förslag' ],
  group: 'digitalungdom',
  usage: 'idea <idea>',
  example: 'idea kommentera er kod bättre',
  serverOnly: false,
  adminOnly: false,
  async execute( message, args ) {
    if ( args.length === 0 ) return message.reply( 'Du måste skicka med ett kort meddelande.' );
    const idea = args.join( ' ' );
    const authorID = message.author.id;
    const authorUsername = message.author.username;
    let id = await getUserByDiscordId( authorID );
    if ( id ) id = id._id;

    await db.collection( 'notifications' ).insertOne( {
      'type': 'idea',
      'where': 'discord',
      'message': idea,
      'author': {
        'id': id,
        'discordID': authorID,
        'discordUsername': authorUsername
      }
    } );

    const notification = createNotificationEmbed( 'IDEA', idea, 65397, { 'id': authorID, 'name': authorUsername, 'url': message.author.displayAvatarURL } );
    const notificationChannel = guild.channels.find( ch => ch.name === 'notifications' );
    notificationChannel.send( '@here', { 'embed': notification } );

    return message.reply( 'tack för din medverkan!' ).then( msg => { msg.delete( 10000 ); } );
  },
};

File: commands/digitalungdom/idea.go
package digitalungdom

import (
	"strings"

	"github.com/digitalungdom-se/dub/pkg"
	"go.mongodb.org/mongo-driver/bson"
)

var Idea = pkg.Command{
	Name:        "idea",
	Description: "Föreslå något till Digital Ungdom",
	Aliases:     []string{"förslag"},
	Group:       "digitalungdom",
	Usage:       "idea <idea>",
	Example:     "idea skaffa programerings tutorials",
	ServerOnly:  false,
	AdminOnly:   false,

	Execute: func(ctx *pkg.Context) error {
		if len(ctx.Args) < 3 {
			_, err := ctx.Reply("Du måste ge ett förslag på minst tre ord.")
			return err
		}

		idea := bson.M{
			"type":    "idea",
			"where":   "discord",
			"message": strings.Join(ctx.Args, " "),
			"author":  ctx.Message.Author.ID}

		err := ctx.Server.Database.InsertNotification(idea)
		if err != nil {
			return err
		}

		_, err = ctx.Reply("Du har nu skickat in ditt förslag. Tack för din medverkan!")

		return err
	},
}
